sympy/core/numbers.py

```diff
@@ -768,7 +768,11 @@ class Float(Number):
 
     is_Float = True
 
-    def __new__(cls, num, prec=None):
+    def __new__(cls, num, dps=None, prec=None):
+        if dps and prec is not None:
```

Comment:
And also in the code below, you can just check `prec` instead of `prec is None`, because it will evaluate to True if it is a nonzero integer (and 0 isn't a valid precision).